maskrcnn_benchmark/modeling/backbone/fbnet_builder.py

```diff
@@ -0,0 +1,822 @@
+        y = super().forward(x)
+        if self.res_connect:
+            y += x
+        return y
+
+
+class ChannelShuffle(nn.Module):
+    def __init__(self, groups):
+        super(ChannelShuffle, self).__init__()
+        self.groups = groups
+
+    def forward(self, x):
+        """Channel shuffle: [N,C,H,W] -> [N,g,C/g,H,W] -> [N,C/g,g,H,w] -> [N,C,H,W]"""
+        N, C, H, W = x.size()
+        g = self.groups
+        assert C % g == 0, "Incompatible group size {} for input channel {}".format(
+            g, C
+        )
+        return (
+            x.view(N, g, int(C / g), H, W)
+            .permute(0, 2, 1, 3, 4)
+            .contiguous()
+            .view(N, C, H, W)
+        )
+
+
+class ConvBNRelu(nn.Sequential):
+    def __init__(
+        self,
+        input_depth,
+        output_depth,
+        kernel,
+        stride,
+        pad,
+        no_bias,
+        use_relu,
+        bn_type,
+        group=1,
+        *args,
+        **kwargs
+    ):
+        super(ConvBNRelu, self).__init__()
+
+        assert use_relu in ["relu", None]
+        if isinstance(bn_type, (list, tuple)):
+            assert len(bn_type) == 2
+            assert bn_type[0] == "gn"
+            gn_group = bn_type[1]
+            bn_type = bn_type[0]
+        assert bn_type in ["bn", "af", "gn", None]
+        assert stride in [1, 2, 4]
+
+        op = Conv2d(
+            input_depth,
+            output_depth,
+            kernel_size=kernel,
+            stride=stride,
+            padding=pad,
+            bias=not no_bias,
+            groups=group,
+            *args,
+            **kwargs
+        )
+        nn.init.kaiming_normal_(op.weight, mode="fan_out", nonlinearity="relu")
+        if op.bias is not None:
+            nn.init.constant_(op.bias, 0.0)
+        self.add_module("conv", op)
+
+        if bn_type == "bn":
+            bn_op = BatchNorm2d(output_depth)
+        elif bn_type == "gn":
+            bn_op = nn.GroupNorm(num_groups=gn_group, num_channels=output_depth)
+        elif bn_type == "af":
+            bn_op = FrozenBatchNorm2d(output_depth)
+        if bn_type is not None:
+            self.add_module("bn", bn_op)
+
+        if use_relu == "relu":
+            self.add_module("relu", nn.ReLU(inplace=True))
+
+
+class SEModule(nn.Module):
+    reduction = 4
+
+    def __init__(self, C):
+        super(SEModule, self).__init__()
+        mid = max(C // self.reduction, 8)
+        conv1 = Conv2d(C, mid, 1, 1, 0)
+        conv2 = Conv2d(mid, C, 1, 1, 0)
+
+        self.op = nn.Sequential(
+            nn.AdaptiveAvgPool2d(1), conv1, nn.ReLU(inplace=True), conv2, nn.Sigmoid()
+        )
+
+    def forward(self, x):
+        return x * self.op(x)
+
+
+class Upsample(nn.Module):
+    def __init__(self, scale_factor, mode):
+        super(Upsample, self).__init__()
+        self.scale = scale_factor
+        self.mode = mode
+
+    def forward(self, x):
+        return interpolate(x, scale_factor=self.scale, mode=self.mode)
```

Comment:
nit: I think we need to pass the `align_corners` here or else we will have a deprecation warning

Reply:
The mode we used ("nearest") requires setting align_corners=None otherwise it will throw an exception.